sdk/src/test/java/io/radar/sdk/util/RadarLogBufferTest.kt

```diff
@@ -0,0 +1,154 @@
+package io.radar.sdk.util
+
+import android.content.Context
+import android.os.Build
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import io.radar.sdk.Radar
+import io.radar.sdk.matchers.RangeMatcher.Companion.isBetween
+import io.radar.sdk.matchers.RangeMatcher.Companion.isGreaterThan
+import io.radar.sdk.model.RadarLog
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertThat
+import org.junit.Assert.assertTrue
+import org.junit.Assume
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.android.util.concurrent.InlineExecutorService
+import org.robolectric.annotation.Config
+import java.io.File
+import java.util.*
+import kotlin.random.Random
+
+@RunWith(AndroidJUnit4::class)
+@Config(sdk=[Build.VERSION_CODES.P])
+class RadarLogBufferTest {
+
+    private val context: Context = ApplicationProvider.getApplicationContext()
+    private val directory = context.getDir("RadarLogs", Context.MODE_PRIVATE)
+
+    @Test
+    fun testLifecycle() {
+        //before the buffer is created, ensure this test begins with an empty file directory
+        assertEquals(0, directory.listFiles()!!.size)
+
+        //create the log buffer
+        val logBuffer = RadarLogBuffer(context, InlineExecutorService())
+
+        //preconditions
+        assertEquals(1, directory.listFiles()!!.size)
+        var logFile = directory.listFiles()!!.first()
+        assertTrue(logFile.exists())
+        var flushable = logBuffer.getLogs()
+        assertTrue(flushable.get().isEmpty())
+        assertEquals(0, logBuffer.size)
+        //a file is added when the logs are retrieved
+        assertEquals(2, directory.listFiles()!!.size)
+
+        //Flush behavior
+        flushable.onFlush(false)
```

Comment:
I like that `onFlush` is explicit, but the trade-off is you have to remember to call it. I wonder, especially given that we're just calling the API after in the SDK, if this would be better encoded as a lambda argument in `logBuffer.getLogs()`

On another note, but `getLogs` seems to imply idempotency to me, rather than actually flushing out the log buffer which is what it's doing. I wonder if we should name it to something like `popLogs` (there is definitely a better name), but that is pretty clear to me that I'm doing something that isn't idempotent.

Reply:
Renamed. I don't think the pattern is 100% ideal, however given the entire log buffer api is internal, I feel like it is acceptable. thoughts?